case_studies/new_detection_encoder/main.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+from os import environ, getenv
+from pathlib import Path
+
+import hydra
+import numpy as np
+import torch
+from einops import rearrange
+from omegaconf import OmegaConf
+from yolov5.models.yolo import DetectionModel
+
+from bliss.models.detection_encoder import DetectionEncoder
+from bliss.train import train
+
+
+class NewDetectionEncoder(DetectionEncoder):
+    def __init__(self, **ode_args):
+        architecture = ode_args.pop("architecture")
+        super().__init__(**ode_args)
+
+        # the checkerboard sizes below only make sense for images composed of an
+        # 8x8 grid of 4x4 tiles. It's straightforward to generalize it to arbitrary
+        # image/tile sizes, but I done so because I haven't determined yet whether
+        # this checkerboard is helpful/necessary
+        shape = (20, 20)
+        base_checkerboard = np.indices(shape).sum(axis=0) % 2
+        self.checkerboard = torch.from_numpy(base_checkerboard).unsqueeze(2).repeat([1, 4, 4])
+        self.checkerboard = self.checkerboard.view(1, 1, 80, 80).float().cuda()
```

Comment:
I'm struggling to understand this checker board and what you are doing here. But maybe you can explain in the meeting. 

Also what do you mean by "images composed of an 8x8 grid of 4x4 tiles."?